cmd/derper/derper.go

```diff
@@ -61,6 +62,7 @@ var (
 	hostname    = flag.String("hostname", "derp.tailscale.com", "LetsEncrypt host name, if addr's port is :443. When --certmode=manual, this can be an IP address to avoid SNI checks")
 	runSTUN     = flag.Bool("stun", true, "whether to run a STUN server. It will bind to the same IP (if any) as the --addr flag value.")
 	runDERP     = flag.Bool("derp", true, "whether to run a DERP server. The only reason to set this false is if you're decommissioning a server but want to keep its bootstrap DNS functionality still running.")
+	secretsURL  = flag.String("secretsURL", "", "SETEC server URL for secrets retrieval of mesh key")
```

Comment:
We conventionally use `snake-case` for flag names (see the examples below).

Optional wording suggestion: `"Secrets server URL (if not set, read the mesh key from a local file)"` or words to that effect.

Reply:
I recommend also adding a flag to specify a secret name prefix, so you can swap to a dev, staging, or other alternative secret value via a flag.

Some notes [here](https://gist.github.com/creachadair/77d57546bf19567531b5a844d9b5452b#adopting-setec) based on my experiences migrating things to setec in the past, some of those might be useful.